### ISSUE TITLE
str.format() doesn't support computed field sizes

### ISSUE THREAD

**Issue by peterhinch** [micropython/micropython#812]
The following produces a syntax error:
"{0:{1}.{1}}".format("rats", 20)

**Comment by pohmelie** [comment]
I implement this feature with +51/-11 lines and will make a pull right after #1749 will be submitted/rejected. The only thing to discuss is fields without name/number. Since I use existing codebase it's hard (or impossible) to make exactly same behaviour as cpython have.
cpython 3.4.3:
``` python
>>> str.format("{:{}{}}", "foo", "^", 10)
'   foo    '
```
micropython
``` python
>>> str.format("{:{}{}}", "^", 10, "foo")
'   foo    '
```
For cpython order is `{0:{1}{2}}`, for micropython — `{2:{0}{1}}`. Micropython do nested first, but cpython take argument right after first '{'. So we can just raise exception on nested fields without names/numbers. Everything else works just fine.

And even better(?):
cpython 3.4.3:
```python
>>> str.format("{text:{align}{widt{h}}} {bar}", text = "foo", align="^", width=10, h="h", bar="hurr")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
ValueError: unexpected '{' in field name
```
micropython
``` python
>>> str.format("{text:{align}{widt{h}}} {bar}", text = "foo", align="^", width=10, h="h", bar="hurr")
'   foo     hurr'
```

**Comment by dpgeorge** [comment]
Now implemented by e3a29de1dc7649656228460caf72e7841440a373.
